processing/src/main/java/org/apache/carbondata/processing/store/writer/v3/CarbonFactDataWriterImplV3.java

```diff
@@ -447,9 +447,7 @@ private void writeDataToFile(FileChannel channel, byte[][] dataChunkBytes) {
         for (int j = 0; j < nodeHolderList.size(); j++) {
           nodeHolder = nodeHolderList.get(j);
           bufferSize = nodeHolder.getDataArray()[i].length;
-          buffer = ByteBuffer.allocate(bufferSize);
-          buffer.put(nodeHolder.getDataArray()[i]);
-          buffer.flip();
+          buffer = ByteBuffer.wrap(nodeHolder.getDataArray()[i]);
```

Comment:
@jackylk If i am not wrong ByteBuffer.wrap could be used even for  below lines of code.
375-377
406-408
441-443